### ISSUE TITLE
NeXT - 01 - Materialize Sequence state

### ISSUE THREAD

**Issue by jtranier** [elaastic/elaastic-questions-server#271]
### What is the state of a sequence?
The state of sequence can be pictured by the steps displayed in the header of the player main pane.
A sequence as a `state`property being one of those value : `beforeShow`, `show`, `afterShow`.
The state of a sequence is also characterized by its `activeInteraction` and the state of this interaction.

### The problem
In order to know the state of a `sequence`, one have to fetch the sequence and its interactions (at least its active one).
This induces more complex queries (join) and lacks of separation of concerns (when I deal with sequence I do not want to bind to interactions).

### The solution
Let's materialize the _sequence state_, meaning storing explicitly this information in the `sequence` table.
Of course, this data introduces a redundancy in the database (but ensure a better separation of concerns and schema).

In order to achieve this, we have to :
- store explicitly the active interaction type (see `InteractionType?`)
- implement the data initialization (deduced from the linked active interaction)
- maintain this state up to date when changing the active interaction of a sequence

### Remarks
- I do not like the terminology  `beforeShow`, `show`, `afterShow` which is much practical than conceptual.
- One can consider that the state of a sequence also comprises the state of all the sequence interactions (one phase being skipped for instance).
- The concept of sequence could be generalized to an arbitrary number of phases, and different types of phases, all those information being part of the sequence state.
- We still have to make the clarification between "phase" and "interaction" (which is quite clear conceptually, but not reflected yet into the code...)

Let's keep in mind those remarks, but stay focus on limited scoped stories being achieved quickly.

### Use sequence state
We need to replace sequence state usage by using the state directly stored on the sequence, instead of considering the bounded interactions :
![Image](https://github.com/user-attachments/assets/c8dfb6c4-188b-4a5b-b772-5a0bcbe3caf7)

Basically, it should allows us to get rid of  `sequenceToUserActiveInteraction`  in `AssignmentOverviewModelFactory`.

### How to validate

- [ ]  Check the various configuration of sequence state represented as icons within the assignment overview is still rendered correctly
- [ ]  Check existing data are properly migrated (i.e. icons remains unchanged for every existing sequences)
- [x]  Check that changing the active interaction of a sequence update accordingly the sequence inner state
